### PR TITLE
content-watcher-api: added validation

### LINKED ISSUE #499: content-watcher-api endpoints data context verification
checking duplicates, incompatible inputs, sanitization,...

### PR DESCRIPTION
**Pull request by aramikm**
# Problem

Adds extra verification on dtos

Close #499

## Steps to Verify:

1. run the api
2. open swagger ui
3. call endpoints with invalid types or amounts or requests

### CODE REVIEW

File: libs/types/src/dtos/content-watcher/subscription.webhook.dto.ts [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
   })
   url: string; // Webhook URL
 
   @IsArray()
+  @ArrayNotEmpty()
+  @IsEnum(AnnouncementTypeName, { each: true })
   @ApiProperty({
     description: 'Announcement types to send to the webhook',
-    example: ['Broadcast', 'Reaction', 'Tombstone', 'Reply', 'Update'],
+    example: ['broadcast', 'reaction', 'tombstone', 'reply', 'update'],
```
Comment (JoeCap08055) [inline libs/types/src/dtos/content-watcher/subscription.webhook.dto.ts:24]:
```suggestion
    isArray: true,
    example: ['broadcast', 'reaction', 'tombstone', 'reply', 'update'],
    enum: AnnouncementTypeName,
    enumName: 'AnnouncementTypeName',
```

**Review by JoeCap08055 — COMMENTED** [review]
I'm thinking, it would make our endpoint validations across the board cleaner, more concise & more readable/understandable if we write some custom validators for commonly used input types, such as:
* MSA ID/Provider ID
* Public control key
* Account address
* Public/private encryption keys
* Block number/block hash

**Review by mattheworris — APPROVED** [review]
Nice work, 🚢 it.

**Comment by aramikm** [comment]
@JoeCap08055 That is a great idea. I've created custom validators for
- Generic Hex values which can be used for public and private keys

I think as you said we can add more customized ones. The issue is that if I add it in here then I've have to decoflict with other open PRs. I'll add this in the last PR so we can go forward with less conflicts.

**Pull request merged**